substrate/frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -2467,6 +2490,29 @@ mod tests {
 		})
 	}
 
+	#[test]
+	fn try_elect_multi_page_fails() {
+		ExtBuilder::default().onchain_fallback(false).build_and_execute(|| {
+			roll_to_signed();
+			assert!(Snapshot::<Runtime>::get().is_some());
+
+			// submit solution and assert it is queued and ready for elect to be called.
+			let (solution, _, _) = MultiPhase::mine_solution().unwrap();
+			assert_ok!(MultiPhase::submit(
+				crate::mock::RuntimeOrigin::signed(99),
+				Box::new(solution),
+			));
+			roll_to(30);
+			assert!(QueuedSolution::<Runtime>::get().is_some());
+
+			// single page elect call works as expected.
+			assert_ok!(MultiPhase::elect(SINGLE_PAGE));
```

Comment:
If you call `MultiPhase::elect(SINGLE_PAGE)` twice in a row, it will fail anyways. So the second call is not being treated fairly, you should first assert that `MultiPhase::elect(10)` is a noop and returns an error, then try `MultiPhase::elect(SINGLE_PAGE)`. Or, do the `MultiPhase::elect(SINGLE_PAGE)` first, but in a transactional layer and rollback.